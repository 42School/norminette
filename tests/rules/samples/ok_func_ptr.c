int			(*f2(void))(int);
int			(*g_fp(int));
int			(*g_v)(int a);
int			(*g_f)(int a, float b);
typedef int	(*t_funcptr)(void);
typedef int	(*t_funcptr)(); //doesnt work

int	main(void)
{
	(*func_pointer)(arg1, arg2);
}
