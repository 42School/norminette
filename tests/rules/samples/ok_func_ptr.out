[36mok_func_ptr.c[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 1":
		<INT> <TAB> <TAB> <TAB> <LPARENTHESIS> <MULT> <IDENTIFIER=f2> <LPARENTHESIS> <VOID> <RPARENTHESIS> <RPARENTHESIS> <LPARENTHESIS> <INT> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_func_ptr.c[0m - [32mIsVarDeclaration[0m In "GlobalScope" from "None" line 2":
		<INT> <TAB> <TAB> <TAB> <LPARENTHESIS> <MULT> <IDENTIFIER=g_fp> <LPARENTHESIS> <INT> <RPARENTHESIS> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_func_ptr.c[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 3":
		<INT> <TAB> <TAB> <TAB> <LPARENTHESIS> <MULT> <IDENTIFIER=g_v> <RPARENTHESIS> <LPARENTHESIS> <INT> <SPACE> <IDENTIFIER=a> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_func_ptr.c[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 4":
		<INT> <TAB> <TAB> <TAB> <LPARENTHESIS> <MULT> <IDENTIFIER=g_f> <RPARENTHESIS> <LPARENTHESIS> <INT> <SPACE> <IDENTIFIER=a> <COMMA> <SPACE> <FLOAT> <SPACE> <IDENTIFIER=b> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_func_ptr.c[0m - [32mIsUserDefinedType[0m In "GlobalScope" from "None" line 5":
		<TYPEDEF> <SPACE> <INT> <TAB> <LPARENTHESIS> <MULT> <IDENTIFIER=t_funcptr> <RPARENTHESIS> <LPARENTHESIS> <VOID> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_func_ptr.c[0m - [32mIsUserDefinedType[0m In "GlobalScope" from "None" line 6":
		<TYPEDEF> <SPACE> <INT> <TAB> <LPARENTHESIS> <MULT> <IDENTIFIER=t_funcptr> <RPARENTHESIS> <LPARENTHESIS> <RPARENTHESIS> <SEMI_COLON> <SPACE> 
[36mok_func_ptr.c[0m - [32mIsComment[0m In "GlobalScope" from "None" line 6":
		<COMMENT=//doesnt work> <NEWLINE>
[36mok_func_ptr.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 7":
		<NEWLINE>
[36mok_func_ptr.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 8":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mok_func_ptr.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 9":
		<LBRACE> <NEWLINE>
[36mok_func_ptr.c[0m - [32mIsFunctionCall[0m In "Function" from "GlobalScope" line 10":
		<TAB> <LPARENTHESIS> <MULT> <IDENTIFIER=func_pointer> <RPARENTHESIS> <LPARENTHESIS> <IDENTIFIER=arg1> <COMMA> <SPACE> <IDENTIFIER=arg2> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_func_ptr.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 11":
		<RBRACE> <NEWLINE>
ok_func_ptr.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	Missing or invalid 42 header
Notice: GLOBAL_VAR_DETECTED  (line:   2, col:   1):	Global variable present in file. Make sure it is a reasonable choice.
